File: src/main/java/dkoryakin/aston/demo/api/body/AccountWithdrawPostBody.java
package dkoryakin.aston.demo.api.body;

import dkoryakin.aston.demo.infrastructure.validator.PinCodeConstraint;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AccountWithdrawPostBody {

    @PinCodeConstraint
    private String pin;

    public Double amount;

}

File: src/main/java/dkoryakin/aston/demo/api/body/AccountWithdrawPostBody.kt
package dkoryakin.aston.demo.api.body

import dkoryakin.aston.demo.infrastructure.validator.PinCodeConstraint
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import org.hibernate.validator.constraints.Length


data class AccountWithdrawPostBody(@NotNull val amount:Double,
                                   @PinCodeConstraint val pin: String) {
}